main.tf

```diff
@@ -1,132 +1,53 @@
 locals {
-  connect_vcs_repo = var.repository_identifier != null ? { create = true } : {}
-  enable_oidc      = var.auth_method == "iam_role_oidc" && var.oidc_settings != null
+  enable_oidc = var.auth_method == "iam_role_oidc" && var.oidc_settings != null
 }
 
 ################################################################################
 # Workspace
 ################################################################################
 
-resource "tfe_workspace" "default" {
-  name                      = var.name
-  allow_destroy_plan        = var.allow_destroy_plan
-  assessments_enabled       = var.assessments_enabled
-  auto_apply                = var.auto_apply
-  auto_apply_run_trigger    = var.auto_apply_run_trigger
-  description               = var.description
-  file_triggers_enabled     = var.file_triggers_enabled
-  global_remote_state       = var.global_remote_state
-  organization              = var.terraform_organization
-  project_id                = var.project_id
-  queue_all_runs            = var.queue_all_runs
-  remote_state_consumer_ids = var.remote_state_consumer_ids
-  ssh_key_id                = var.ssh_key_id
-  tag_names                 = var.workspace_tags
-  terraform_version         = var.terraform_version
-  trigger_patterns          = var.trigger_patterns
-  trigger_prefixes          = var.trigger_prefixes
-  working_directory         = var.working_directory
-
-  dynamic "vcs_repo" {
-    for_each = local.connect_vcs_repo
-
-    content {
-      identifier         = var.repository_identifier
-      branch             = var.branch
-      ingress_submodules = false
-      oauth_token_id     = var.oauth_token_id
-    }
-  }
-}
-
-resource "tfe_workspace_settings" "default" {
-  agent_pool_id  = var.execution_mode == "agent" ? var.agent_pool_id : null
-  execution_mode = var.execution_mode
-  workspace_id   = tfe_workspace.default.id
-}
-
-resource "tfe_notification_configuration" "default" {
-  for_each = var.notification_configuration
-
-  name             = each.key
-  destination_type = each.value.destination_type
-  enabled          = each.value.enabled
-  triggers         = each.value.triggers
-  url              = each.value.url
-  workspace_id     = tfe_workspace.default.id
+module "tfe-workspace" {
+  source  = "schubergphilis/mcaf-workspace/tfe"
+  version = "2.0.0"
```

Comment:
Maybe worth changing to to e.g. `~> 2.0` so new fixes/features (backwards compatible) and patches are implemented automatically?

See also: https://developer.hashicorp.com/terraform/language/expressions/version-constraints